stronk/schemas/exercise/query.py

```diff
@@ -13,17 +16,27 @@ class Query(graphene.ObjectType):
                               author=graphene.String())
 
     def resolve_exercises(root, info, author=None):
-        """Return a list of all exercises."""
+        """Return a list of all exercises.
+
+           Search by author: A user's username.
+        """
+        query = Exercise.get_query(info)
         if author:
+            user = UserModel.find_by_username(author)
+            if not user:
+                raise NotFound(USER_NOT_FOUND_MSG)
+
             return (query
                     .order_by(ExerciseModel.name.desc())
-                    .filter(ExerciseModel.author == author)
+                    .filter(ExerciseModel.author == user.id)
                     .all())
 
-        return Exercise.get_query(info).all()
+        return query.all()
+
+    def resolve_exercise(root, info, id=None, name=None, desc=None):
```

Comment:
Since we dropped the author field, I wonder if we should move name and desc to `resolve_exercises` later as well since it's quite possible they would return multiple exercises?